frb_codegen/src/main.rs

```diff
@@ -26,3 +26,185 @@ fn main() -> anyhow::Result<()> {
     info!("Now go and use it :)");
     Ok(())
 }
+
+mod tests {
+    use lazy_static::lazy_static;
+    use lib_flutter_rust_bridge_codegen::init_logger;
+
+    lazy_static! {
+        static ref LOGGER: () = init_logger(".", true).unwrap();
+    }
+
+    // VS Code runs in frb_codegen with "Run test" and flutter_rust_bridge with "Debug test" >_>
+    #[cfg(test)]
+    fn set_dir() {
+        use std::fs;
+
+        if let Ok(metadata) = fs::metadata("frb_codegen") {
+            if metadata.is_dir() {
+                std::env::set_current_dir("frb_codegen").unwrap();
+            }
+        }
+    }
+
+    /// `cargo build` can fail because frb_example/pure_dart makes testing a build step! *gasp*
+    /// If you want to get `cargo build` working again and investigate why pure_dart fails:
+    /// 1. mv ../frb_example/pure_dart/rust/build.rs ../frb_example/pure_dart/rust/_build.rs
+    /// 2. Run this test in the debugger - which is a copy of build.rs (but more complete)
```

Comment:
```suggestion
    /// When the `frb_example/pure_dart` fails to build, i.e. the `cargo build` there fails,
    /// i.e. the frb_codegen fails to generate code in that repo, you may want to use this
    /// test to examine the problems, because this one is a copy of `build.rs` in that
    /// `frb_example/pure_dart`. For example, you may run this `fn pure_dart()` unit
    /// test in the debugger.
    /// 
    /// In some scenarios, such as when using VSCode to execute this test, the `cargo build`
    /// will be run before this `fn pure_dart()` test gets executed (see #1106 for details).
    /// Therefore, you may even fail to execute *this* function. In that case, you may run: 
    /// `mv ../frb_example/pure_dart/rust/build.rs ../frb_example/pure_dart/rust/_build.rs`
    /// Then that `build.rs` is temporarily disabled and cargo build can run.
```